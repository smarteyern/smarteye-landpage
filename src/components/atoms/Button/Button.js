import React from "react";
import propTypes from "prop-types";

import {
  ButtonRoot,
  ButtonOutlined,
  ButtonLink,
  ButtonsVariants,
} from "./styles";

const Button = ({ type, children, color, disabled, onClick, variant }) => (
  <ButtonRoot
    disabled={disabled}
    type={type}
    color={color}
    onClick={onClick}
    variant={variant}
  >
    {children}
  </ButtonRoot>
);

const ButtonWrapper = (props) => {
  switch (props.variant) {
    case ButtonsVariants.outlined:
      return <ButtonOutlined {...props} />;
    case ButtonsVariants.link:
      return <ButtonLink {...props} />;
    default:
      return <Button {...props} />;
  }
};

ButtonWrapper.defaultProps = {
  type: "button",
  children: undefined,
  color: "default",
  variant: "default",
};

ButtonWrapper.propTypes = {
  type: propTypes.string,
  children: propTypes.node,
  color: propTypes.oneOf(["default", "primary", "error"]),
  variant: propTypes.oneOf(["default", "outlined", "link"]),
};

export default ButtonWrapper;
