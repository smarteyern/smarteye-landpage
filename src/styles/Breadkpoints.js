export const BreadkpointsSize = {
  xs: 576,
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200,
  xxl: 1400,
};

export const breakAt = (size) => `@media (min-width: ${size}px)`;
