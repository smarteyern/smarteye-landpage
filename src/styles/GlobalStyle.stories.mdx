import { Meta, Preview, Story } from "@storybook/addon-docs/blocks";
import GlobalStyleComposed from "./GlobalStyle";

<Meta title="Styles/GlobalStyle" component={GlobalStyleComposed} />

# Global Style

CSS Resets stylesheets

```jsx
const App = () => (
  <>
    <GlobalStyle />
    <OtherComponent />
  </>
);
 `;
```

## Typography

Basic global typography.

All HTM headings. `h1` through `h6`, are available:

<Preview>
  <Story name="heading">
    <div>
      <h1>Título 1</h1>
      <h2>Título 2</h2>
      <h3>Título 3</h3>
      <h4>Título 4</h4>
      <h5>Título 5</h5>
      <h6>Título 6</h6>
    </div>
  </Story>
</Preview>

### Paragraph

Styling for common text elements:

<Preview>
  <Story name="Paragraph">
    <div>
      <p>The quick brown fox jumps over the lazy dog.</p>
      <p>
        <strong>The quick brown fox jumps over the lazy dog.</strong>
      </p>
      <p>
        <em>The quick brown fox jumps over the lazy dog.</em>
      </p>
      <p>
        <u>The quick brown fox jumps over the lazy dog.</u>
      </p>
      <p>
        <s>The quick brown fox jumps over the lazy dog.</s>
      </p>
    </div>
  </Story>
</Preview>
