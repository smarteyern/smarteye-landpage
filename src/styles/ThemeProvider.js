import React from "react";
import { ThemeProvider as StyledProvider } from "styled-components";

export const ThemeNames = {
  light: "light",
  dark: "dark",
};

const light = {
  colors: {
    primary: {
      main: "#2D9CDB",
      dark: "#225CAA",
      light: "#56CCF2",
    },
    text: {
      main: "#333333",
      primary: "#4F4F4F",
      secundary: "#BDBDBD",
      light: "#F4F4F4",
    },
    background: {
      main: "#FFFFFF",
      default: "#F2F2F2",
      paper: "#E0E0E0",
      dark: "#2A2A2A",
      darkBlue: "#003557",
    },
    error: {
      main: "#EB5757",
      dark: "#b22a00",
    },
    status: {
      sucess: "#27AE60",
      attention: "#F2C94C",
    },
  },
};

export const allTheme = {
  light,
  dark: {
    ...light,
    colors: {
      ...light.colors,
      primary: {
        main: "#225CAA",
        dark: "#56CCF2",
        light: "#2D9CDB",
      },
      text: {
        main: "#F4F4F4",
        primary: "#F2F2F2",
        secundary: "#BDBDBD",
        light: "#E0E0E0",
      },
      background: {
        default: "#333333",
        paper: "#4F4F4F",
      },
    },
  },
};

const ThemeProvider = ({ themes, children }) => (
  <StyledProvider theme={allTheme[themes]}>{children}</StyledProvider>
);

ThemeProvider.defaultProps = {
  themes: "light",
};

export default ThemeProvider;
