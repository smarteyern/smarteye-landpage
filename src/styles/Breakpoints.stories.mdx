import { Meta } from "@storybook/addon-docs/blocks";
import { BreadkpointsSize } from "./Breadkpoints.js";

<Meta title="Styles/Breakpoints" />

# Breakpoints

Breakpoints for responsives layout:

<table>
  <tr>
    <th>Extra Small (xs)</th>
    <td>Default | menor que {BreadkpointsSize.xs}px</td>
  </tr>
  <tr>
    <th>Small (sm)</th>
    <td> >= {BreadkpointsSize.sm}px</td>
  </tr>
  <tr>
    <th>Medium (md)</th>
    <td> >= {BreadkpointsSize.md}px</td>
  </tr>
  <tr>
    <th>Large (lg)</th>
    <td> >= {BreadkpointsSize.lg}px</td>
  </tr>
  <tr>
    <th>Extra large (xl)</th>
    <td> >= {BreadkpointsSize.xl}px</td>
  </tr>
  <tr>
    <th>Extra extra large (xxl)</th>
    <td> >= {BreadkpointsSize.xxl}px</td>
  </tr>
</table>

```js
import { breakAt, BreadkpointsSize } from "Breackpoints";

const title = styled.h1`
  font-size: 1rem;
  ${breakAt(BreadkpointsSize.md)} {
    font-size: 2rem;
  }
`;
```
