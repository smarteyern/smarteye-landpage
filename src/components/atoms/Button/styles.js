import styled from "styled-components";
import { allTheme } from "styles/ThemeProvider";

export const ButtonColors = {
  default: "default",
  primary: "primary",
  error: "error",
};

export const ButtonsVariants = {
  default: "default",
  outlined: "outlined",
  link: "link",
};

const getColorButton = ({ theme, color }) => {
  switch (color) {
    case ButtonColors.primary:
      return theme.colors.primary.main;
    case ButtonColors.error:
      return theme.colors.error.main;
    default:
      return theme.colors.text.secundary;
  }
};

const getColorDark = ({ theme, color }) => {
  switch (color) {
    case ButtonColors.primary:
      return theme.colors.primary.dark;
    case ButtonColors.error:
      return theme.colors.error.dark;
    default:
      return theme.colors.text.primary;
  }
};

export const ButtonRoot = styled.button`
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: ${getColorButton};
  border: 2px solid ${getColorButton};
  border-radius: 4px;
  color: ${allTheme.light.colors.text.light};

  &:disabled {
    opacity: 0.6;
    cursor: not-allowed;
  }

  &:hover:enabled {
    background-color: ${getColorDark};
    border: 2px solid ${getColorDark};
  }
`;

export const ButtonOutlined = styled(ButtonRoot)`
  background-color: transparent;
  color: ${getColorButton};

  &:hover:enabled {
    background-color: transparent;
    color: ${getColorDark};
    border: 2px solid ${getColorDark};
  }
`;

export const ButtonLink = styled(ButtonRoot)`
  background-color: transparent;
  border-color: transparent;
  color: ${getColorButton};
  padding-left: 0;
  padding-right: 0;

  &:hover:enabled {
    background-color: transparent;
    border-color: transparent;
    color: ${getColorDark};
  }
`;

export const Toolbar = styled.div`
  & > * {
    margin-right: 8px;
  }
`;
