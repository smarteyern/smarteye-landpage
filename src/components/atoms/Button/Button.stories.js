import React from "react";
import { actions } from "@storybook/addon-actions";
import Button from "./Button";
import { Toolbar } from "./styles";

export default {
  title: "Components/Atoms/Button",
  component: Button,
};

const events = actions({ onClick: "clicked" });

export const usage = () => (
  <div>
    <p>Enabled:</p>
    <Toolbar>
      <Button {...events} color="default">
        Default
      </Button>
      <Button color="primary" {...events}>
        Primary
      </Button>
      <Button color="error" {...events}>
        Error
      </Button>
    </Toolbar>
    <p>Desabled:</p>
    <Toolbar>
      <Button disabled {...events} color="default">
        Default
      </Button>
      <Button disabled color="primary" {...events}>
        Primary
      </Button>
      <Button disabled color="error" {...events}>
        Error
      </Button>
    </Toolbar>
  </div>
);

export const outlined = () => (
  <div>
    <p>Enabled:</p>
    <Toolbar>
      <Button {...events} variant="outlined">
        Default
      </Button>
      <Button color="primary" {...events} variant="outlined">
        Primary
      </Button>
      <Button color="error" {...events} variant="outlined">
        Error
      </Button>
    </Toolbar>
    <p>Desabled:</p>
    <Toolbar>
      <Button disabled {...events} variant="outlined">
        Default
      </Button>
      <Button disabled color="primary" {...events} variant="outlined">
        Primary
      </Button>
      <Button disabled color="error" {...events} variant="outlined">
        Error
      </Button>
    </Toolbar>
  </div>
);

export const link = () => (
  <div>
    <p>Enabled:</p>
    <Toolbar>
      <Button {...events} variant="link">
        Default
      </Button>
      <Button color="primary" {...events} variant="link">
        Primary
      </Button>
      <Button color="error" {...events} variant="link">
        Error
      </Button>
    </Toolbar>
    <p>Desabled:</p>
    <Toolbar>
      <Button disabled {...events} variant="link">
        Default
      </Button>
      <Button disabled color="primary" {...events} variant="link">
        Primary
      </Button>
      <Button disabled color="error" {...events} variant="link">
        Error
      </Button>
    </Toolbar>
  </div>
);
