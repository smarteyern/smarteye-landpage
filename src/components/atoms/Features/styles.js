import styled from "styled-components";

import { breakAt, BreadkpointsSize } from "styles/Breadkpoints";

const getPrimaryDark = ({ theme }) => theme.colors.primary.dark;
const getPrimaryLight = ({ theme }) => theme.colors.primary.light;
const getPrimaryText = ({ theme }) => theme.colors.text.light;

export const Root = styled.div`
  text-align: center;

  & h5 {
    margin: 16px 0;
    min-height: 3.12rem;

    ${breakAt(BreadkpointsSize.lg)} {
      min-height: 3.9rem;
    }
  }

  & p {
    margin: 0;
  }
`;

export const IconRoot = styled.div`
  display: inline-block;
  align-items: center;
  align-content: center;
  width: 116px;
  height: 116px;
  padding: 8px;
  border-radius: 50%;

  &:hover {
    border: 1px solid ${getPrimaryDark};

    > * {
      color: ${getPrimaryText};
      background-color: ${getPrimaryDark};
    }
  }
`;

export const IconContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 32px;
  width: 100%;
  height: 100%;
  border: 1px solid ${getPrimaryLight};
  border-radius: 50%;
  color: ${getPrimaryLight};
`;
