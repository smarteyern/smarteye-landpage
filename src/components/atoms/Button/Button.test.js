import React from "react";
import { render, fireEvent } from "test-utils";
import Button from "./Button";
import { ButtonColors, ButtonsVariants } from "./styles";

test("Renders a text", () => {
  const { getByText } = render(<Button>Click Here</Button>);

  expect(getByText("Click Here")).toBeInTheDocument();
});

test("Trigger event on click", () => {
  const handleClick = jest.fn();

  const { getByRole } = render(<Button onClick={handleClick} />);
  fireEvent.click(getByRole("button"));

  expect(handleClick).toBeCalled();
});

test.each(Object.values(ButtonColors).map((item) => [item]))(
  "Renders with color %s",
  (color) => {
    const { asFragment } = render(<Button color={color} />);

    expect(asFragment()).toMatchSnapshot();
  }
);

test.each(Object.values(ButtonsVariants).map((item) => [item]))(
  "Renders with variants %s",
  (variant) => {
    const { asFragment } = render(<Button variant={variant} />);

    expect(asFragment()).toMatchSnapshot();
  }
);

test("Renders with primary and variants %s", () => {
  const { asFragment } = render(<Button variant="outlined" color="primary" />);

  expect(asFragment()).toMatchSnapshot();
});

test("Renders with link and variants %s", () => {
  const { asFragment } = render(<Button variant="link" color="primary" />);

  expect(asFragment()).toMatchSnapshot();
});
