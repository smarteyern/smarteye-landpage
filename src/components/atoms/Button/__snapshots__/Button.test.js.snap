// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['Renders with color default 1'] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #BDBDBD;
  border: 2px solid #BDBDBD;
  border-radius: 4px;
  color: #F4F4F4;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #4F4F4F;
  border: 2px solid #4F4F4F;
}

<button
    class="c0"
    color="default"
    type="button"
  />
</DocumentFragment>
`;

exports['Renders with color error 1'] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #EB5757;
  border: 2px solid #EB5757;
  border-radius: 4px;
  color: #F4F4F4;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #b22a00;
  border: 2px solid #b22a00;
}

<button
    class="c0"
    color="error"
    type="button"
  />
</DocumentFragment>
`;

exports['Renders with color primary 1'] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #2D9CDB;
  border: 2px solid #2D9CDB;
  border-radius: 4px;
  color: #F4F4F4;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #225CAA;
  border: 2px solid #225CAA;
}

<button
    class="c0"
    color="primary"
    type="button"
  />
</DocumentFragment>
`;

exports['Renders with link and variants %s 1'] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #2D9CDB;
  border: 2px solid #2D9CDB;
  border-radius: 4px;
  color: #F4F4F4;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #225CAA;
  border: 2px solid #225CAA;
}

.c1 {
  background-color: transparent;
  border-color: transparent;
  color: #2D9CDB;
  padding-left: 0;
  padding-right: 0;
}

.c1:hover:enabled {
  background-color: transparent;
  border-color: transparent;
  color: #225CAA;
}

<button
    class="c0 c1"
    color="primary"
    type="button"
  />
</DocumentFragment>
`;

exports['Renders with primary and variants %s 1'] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #2D9CDB;
  border: 2px solid #2D9CDB;
  border-radius: 4px;
  color: #F4F4F4;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #225CAA;
  border: 2px solid #225CAA;
}

.c1 {
  background-color: transparent;
  color: #2D9CDB;
}

.c1:hover:enabled {
  background-color: transparent;
  color: #225CAA;
  border: 2px solid #225CAA;
}

<button
    class="c0 c1"
    color="primary"
    type="button"
  />
</DocumentFragment>
`;

exports['Renders with variants default 1'] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #BDBDBD;
  border: 2px solid #BDBDBD;
  border-radius: 4px;
  color: #F4F4F4;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #4F4F4F;
  border: 2px solid #4F4F4F;
}

<button
    class="c0"
    color="default"
    type="button"
  />
</DocumentFragment>
`;

exports['Renders with variants link 1'] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #BDBDBD;
  border: 2px solid #BDBDBD;
  border-radius: 4px;
  color: #F4F4F4;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #4F4F4F;
  border: 2px solid #4F4F4F;
}

.c1 {
  background-color: transparent;
  border-color: transparent;
  color: #BDBDBD;
  padding-left: 0;
  padding-right: 0;
}

.c1:hover:enabled {
  background-color: transparent;
  border-color: transparent;
  color: #4F4F4F;
}

<button
    class="c0 c1"
    color="default"
    type="button"
  />
</DocumentFragment>
`;

exports['Renders with variants outlined 1'] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #BDBDBD;
  border: 2px solid #BDBDBD;
  border-radius: 4px;
  color: #F4F4F4;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #4F4F4F;
  border: 2px solid #4F4F4F;
}

.c1 {
  background-color: transparent;
  color: #BDBDBD;
}

.c1:hover:enabled {
  background-color: transparent;
  color: #4F4F4F;
  border: 2px solid #4F4F4F;
}

<button
    class="c0 c1"
    color="default"
    type="button"
  />
</DocumentFragment>
`;
